[Backport release-25.05] geoserver: 2.27.0 -> 2.27.1

mdaniels5757 opened the pull request:
Manual backport of #415077 to `release-25.05`.

- [x] Before merging, ensure that this backport is [acceptable for the release](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#changes-acceptable-for-releases).
    * Even as a non-committer, if you find that it is not acceptable, leave a comment.

r-ryantm committed:
geoserver: 2.27.0 -> 2.27.1

(cherry picked from commit e64838938a6f8cc0620ef6c25161d3fb9c2b1fe8)

sikmir committed:
geoserver: modernize

(cherry picked from commit 8d8285253bb1933273d011c85fb8bddb5407be37)

mdaniels5757 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 419147 `

Logs: https://github.com/mdaniels5757/nixpkgs-review-gha/actions/runs/15813721020


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>geoserver</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>geoserver</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>geoserver</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>geoserver</li>
  </ul>
</details>

mdaniels5757 commented:
> This report is automatically generated by the `check-cherry-picks` CI workflow.
> ...

Yes, this was due to merge conflicts.

imincik reviewed (dismissed):
Great thanks @mdaniels5757 !

imincik reviewed (changes requested):
Great thanks for backport @mdaniels5757 . I requested one small change.

mdaniels5757 committed:
geoserver: remove rec when using finalAttrs

Not a cherry-pick, but identical to the changes to this file
in the treewide d75957a3bbc310475425c9042174225c837f9473.

imincik reviewed (approved):
Looks good. Thank you very much.
